src/pipecat/services/gemini_multimodal_live/gemini_vertex.py

```diff
@@ -0,0 +1,222 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+"""Google Gemini Multimodal Live API service on Vertex AI implementation.
+
+This module provides real-time conversational AI capabilities using Google's
+Gemini Multimodal Live API, supporting both text and audio modalities with
+voice transcription, streaming responses, and tool usage.
+"""
+
+import inspect
+import json
+import os
+import re
+from typing import Optional
+
+from loguru import logger
+from websockets.asyncio.client import connect as websocket_connect
+
+from pipecat.frames.frames import (
+    EndFrame,
+    ErrorFrame,
+    Frame,
+    LLMFullResponseStartFrame,
+    LLMMessagesAppendFrame,
+    LLMTextFrame,
+    StartFrame,
+    TTSAudioRawFrame,
+    TTSStartedFrame,
+    TTSStoppedFrame,
+    TTSTextFrame,
+)
+from pipecat.services.gemini_multimodal_live.gemini import (
+    GeminiMultimodalLiveLLMService,
+    GeminiMultimodalModalities,
+)
+from pipecat.services.gemini_multimodal_live.gemini import (
+    InputParams as GeminiInputParams,
+)
+from pipecat.services.openai.llm import OpenAILLMService
+
+try:
+    from google.auth import default
+    from google.auth.exceptions import GoogleAuthError
+    from google.auth.transport.requests import Request
+    from google.oauth2 import service_account
+    from openai._types import NOT_GIVEN, NotGiven
+
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error(
+        "In order to use Google AI, you need to `pip install pipecat-ai[google]`. Also, set `GOOGLE_APPLICATION_CREDENTIALS` environment variable."
+    )
+    raise Exception(f"Missing module: {e}")
+
+
+class GeminiMultimodalLiveVertexLLMService(GeminiMultimodalLiveLLMService):
+    """Provides access to Google's Gemini Multimodal Live API on Vertex AI.
+
+    This service enables real-time conversations with Gemini, supporting both
+    text and audio modalities. It handles voice transcription, streaming audio
+    responses, and tool usage. It does this for Vertex AI compatible models.
+    """
+
+    class VertexInputParams(OpenAILLMService.InputParams, GeminiInputParams):
+        """Input parameters specific to Vertex AI.
+
+        Parameters:
+            location: GCP region for Vertex AI endpoint (e.g., "us-east4" or "us-central1").
+            project_id: Google Cloud project ID.
+        """
+
+        # https://cloud.google.com/vertex-ai/generative-ai/docs/learn/locations
+        location: str = "us-central1"
+        project_id: str
+
+    def __init__(
+        self,
+        *,
+        api_key: str = "ignore",
```

Comment:
why do we need this?

Reply:
because base class `GeminiMultimodalLiveLLMService` requires an `api_key`.
It's not pretty, but otherwise an api_key needs to be passed in by user, and that feels wrong since it is not used.